src/script/script.h

```diff
@@ -571,16 +571,34 @@ class CScript : public CScriptBase
         if (b.empty())
             return nFound;
         iterator pc = begin();
+        iterator copied = begin();
+        size_t nShift = 0;
         opcodetype opcode;
         do
         {
             while (end() - pc >= (long)b.size() && memcmp(&pc[0], &b[0], b.size()) == 0)
             {
-                pc = erase(pc, pc + b.size());
+                // Lazy copy-in-place if there is a match:
+                while (copied != pc-nShift) {
```

Comment:
A test for !nShift here would prevent copying everything upto the first match to itself